pkg/resource/deploy/builtins.go

```diff
@@ -209,6 +210,11 @@ func (p *builtinProvider) readStackReference(inputs resource.PropertyMap) (resou
 		}
 	}
 
+	// Sort the secret outputs
```

Comment:
Say why we're sorting? Maybe something like the following?

```suggestion
	// Sort the secret outputs so the order is deterministic, to avoid spurious diffs during updates.
```

Reply:
Yep, this is much better!